compiler/parser/RdrHsSyn.hs

```diff
@@ -3018,9 +3018,9 @@ mkChoiceByKeyInstanceDecl templateName ValidTemplate{..} CombinedChoiceData { cc
 mkInterfaceFixedChoiceInstanceDecl :: Located RdrName -> InterfaceChoiceSignature -> [LHsDecl GhcPs]
 mkInterfaceFixedChoiceInstanceDecl tycon InterfaceChoiceSignature {..} =
   [ mkInstance "HasToAnyChoice"
-      (mkTemplateClassMethod "_toAnyChoice" [proxy] (mkPrimitive "primitive" "EToAnyInterfaceChoice" `mkApp` nil ifaceType) Nothing)
+      (mkTemplateClassMethod "_toAnyChoice" [proxy] (mkPrimitive "primitive" "EToAnyChoice" `mkApp` nil ifaceType) Nothing)
   , mkInstance "HasFromAnyChoice"
-      (mkTemplateClassMethod "_fromAnyChoice" [proxy] (mkPrimitive "primitive" "EFromAnyInterfaceChoice" `mkApp` nil ifaceType) Nothing)
+      (mkTemplateClassMethod "_fromAnyChoice" [proxy] (mkPrimitive "primitive" "EFromAnyChoice" `mkApp` nil ifaceType) Nothing)
```

Comment:
For symmetry,

```suggestion
      (mkTemplateClassMethod "_fromAnyChoice" [] (mkPrimitive "primitive" "EFromAnyChoice") Nothing)
```

diff:

```diff
-   _fromAnyChoice _ = GHC.Types.primitive @"EFromAnyChoice" ([] : [MyInterface])
+   _fromAnyChoice = GHC.Types.primitive @"EFromAnyChoice"
```

After these changes, the local variables `proxy` and `nil` from the where clause will be redundant.